webapp/src/components/issue_attribute_selector.jsx

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import ReactSelect from 'react-select';
+import PropTypes from 'prop-types';
+
+import {getStyleForReactSelect} from 'utils/styles';
+import Setting from 'components/setting';
+
+export default class IssueAttributeSelector extends PureComponent {
+    static propTypes = {
+        name: PropTypes.string,
+        repo: PropTypes.string,
+        required: PropTypes.bool,
+        isMulti: PropTypes.bool,
+        resetInvalidOnChange: PropTypes.bool,
+        addValidate: PropTypes.func,
+        removeValidate: PropTypes.func,
+        onChange: PropTypes.func,
+        load: PropTypes.func,
+        theme: PropTypes.object.isRequired,
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            options: [],
+            values: [],
+            isLoading: false,
+            invalid: false,
+            error: '',
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.addValidate) {
+            this.props.addValidate(this.isValid);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.removeValidate) {
+            this.props.removeValidate(this.isValid);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && this.props.value !== prevProps.value) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
+
+        if (prevProps.repo !== this.props.repo) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions = async () => {
+        this.setState({isLoading: true});
+        const options = await this.props.load();
+
+        // filter out currently selected options that do not exist in the new repo
+        const optionValues = options.map((option) => option.value);
+        const validValues = this.state.values.filter((value) => optionValues.includes(value.value));
+
+        this.setState({
+            options,
+            values: validValues,
+            isLoading: false,
+        });
+    }
+
+    onChange = (values) => {
+        this.setState({values});
+
+        if (!values) {
+            this.props.onChange(this.props.isMulti ? [] : '');
+            return;
+        }
+
+        this.props.onChange(this.props.isMulti ? values.map((v) => v.value) : values.value);
+
+        if (this.props.resetInvalidOnChange) {
+            this.setState({invalid: false});
+        }
+    };
+
+    isValid = () => {
+        if (!this.props.required) {
+            return true;
+        }
+
+        const valid = Boolean(this.props.value && this.props.value.toString().length !== 0);
```

Comment:
`this.props.value.toString().length` is a clever way to check for presence, but I would like this to be a bit more explicit. The fact that `[].toString() === ''` is not something that's intuitive. Does the following work?

```suggestion
        const valid = Boolean(this.props.value && this.props.value.length);
```